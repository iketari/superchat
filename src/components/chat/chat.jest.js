import Chat from './chat';

let chat;

beforeEach(() => {
	chat = new Chat({
		el: document.createElement('div'),
		data: { messages: [] },
		avatarService: {
			getAvatar() {
				return 'avatar.jpg';
			}
		}
	});
});

test('Should render a banner for empty message list', () => {
	chat.render();

	expect(chat.el.innerHTML).toMatchSnapshot();
});

test('Should render one message', () => {
	const messages = [
		{
			name: 'name',
			text: 'text',
			date: 100500
		}
	]
	chat.add(messages);

	chat.render();

	expect(chat.el.innerHTML).toMatchSnapshot();
});