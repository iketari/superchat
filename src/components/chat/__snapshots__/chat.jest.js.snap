// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a banner for empty message list 1`] = `
"
<div class=\\"chat\\">
  <div class=\\"chat__container\\">
    <div class=\\"header\\">
      <h2>You are logged in as </h2><small><a href=\\"/main\\">To main menu</a></small>
    </div>
    <div class=\\"chat__box\\">
      <h3>Пока нет сообщений</h3>
    </div>
  </div>
</div>"
`;

exports[`Should render one message 1`] = `
"
<div class=\\"chat\\">
  <div class=\\"chat__container\\">
    <div class=\\"header\\">
      <h2>You are logged in as </h2><small><a href=\\"/main\\">To main menu</a></small>
    </div>
    <div class=\\"chat__box\\">
      <div class=\\"message-box right-img\\">
        <div class=\\"picture\\"><img src=\\"avatar.jpg\\" title=\\"name of user\\"><span class=\\"time\\">00:01:40</span></div>
        <div class=\\"message\\"><span>name</span>
          <p>text</p>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
